ci/download_dependencies.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+set -euxo pipefail
+
+source "${WORKDIR}/ci/archs.sh"
+
+temp_dir="${WORKDIR}/bin/deps/releases"
+lib_root="${WORKDIR}/bin/deps/lib"
+
+libtelio_version="v5.0.0-rc2"
```

Comment:
Are those a single places where version numbers will be defined? Or they have to be specified somewhere else also?

Reply:
Ahh, yes, that's the single place. I should probably put that somewhere more accessible :thinking: , maybe `.env`?

Reply:
Update: this will need to change anyway because of the libtelio integration in their pipelines - there is a need to be able to control the version used, so I'll definitely change it

Reply:
I think it is ok to have it here. But wouldn't it need to be synchronized with submodules commit?

Reply:
:thinking:, yes, you are right. So there are more things which need to be synchronized.

For internal build:
- version/tag/sha of the release so we have correct .so files for internal builds
- version/tag/sha of the go bindings for all libs

For public build
- version/tag/sha of the libdrop and libtelio for the public builds
- version/tag/sha of the go bindings for all libs

And those need to be controlled by env variables.
Ok, so this needs more work

Reply:
Ok I did some updates. Now, the library versions are defined in `lib-versions.env`. For now it's just the versions used to download .so files. I left the comment there to make sure that when someone changes those versions to also take into account the bindings, but for now I won't implement setting the same version for bindings, because it would require dirty hacks with patch files (same way we have with adding moose to go.mod). I'm not yet convinced we need this, but I'll think about that more.

Regarding the `export_lib_versions.sh` - it will be used by the CI to export the versions defined there but still allows to override them by the pipeline - so we have a default, but we can override it still. This will be helpful for libtelio team's integration pipeline